cpusupport/Build/cpusupport-ARM-CRC32_64.c

```diff
@@ -0,0 +1,22 @@
+#ifdef __ARM_ACLE
+#include <arm_acle.h>
+
+#ifndef __ARM_FEATURE_CRC32
+#error "ARM_FEATURE_CRC32 not supported"
+#endif
+
+int
+main(void)
+{
+	uint32_t state = 0;
+	uint8_t x = 0;
+	uint8_t buf[8] = {0, 0, 0, 0, 0, 0, 0, 0};
+
+	/* Test 8-bit and 64-bit. */
+	state = __crc32b(state, x);
+	state = __crc32cd(state, *(const uint64_t *)&buf[0]);
+
+	return (0);
+}
+
+#endif /* !__ARM_ACLE */
```

Comment:
Shouldn't this be at the top?  I mean, it makes more sense to have this file fail to compile because `__crc32b` isn't defined on a compiler which is unaware of the extensions than than to fail to link because `main` isn't defined.